fleetspeak/src/client/https/streaming.go

```diff
@@ -337,7 +347,7 @@ func (c *connection) groupMessages(ctx context.Context) []comms.MessageInfo {
 	ctx, fin := context.WithTimeout(ctx, time.Second)
 	defer fin()
 	for {
-		if size >= sendBytesThreshold || len(r) >= sendCountThreshold {
+		if size >= sendBytesThreshold/2 || len(r) >= sendCountThreshold {
```

Comment:
That parameter was set with the non-streaming use case in mind (and still used by the non-streaming communicator). In that case, we wait for the response before sending more so we might as well send lots - near the maximum buffer size that we and the server want to deal with. In the streaming case, we can have multiple outstanding message blocks out, so we might as well break them up a bit more to set up a bit of a pipeline.

Is totally a bit of a guess though, I haven't experimented to see what is actually optimal. It might also be worth a comment. Let me think about how to phrase it.